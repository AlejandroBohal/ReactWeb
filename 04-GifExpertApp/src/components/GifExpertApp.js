import React, { useState } from 'react'
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = ({defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories);

    //const handleAdd = (category) =>{
        //setCategories(categories => [...categories,'category']);
    //    setCategories([...categories,category]);
    //}
    return (
        <>
          <h2>GifExpertApp</h2>
          <AddCategory setCategories={setCategories}></AddCategory>
          <hr/>
          <ol>
              { 
                categories.map(category => (
                    //<li key={category}>{category}</li>
                    <GifGrid
                        key={category}
                        category={category}>
                        
                    </GifGrid>)
                ) 
              }
          </ol>
        </>
    )
}

export default GifExpertApp
