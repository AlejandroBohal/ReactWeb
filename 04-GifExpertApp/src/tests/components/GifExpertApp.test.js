import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../../components/GifExpertApp';

describe('Testing GifExpertApp component', () => {
    let wrapper = shallow(<GifExpertApp />);
    beforeEach( () =>{
        wrapper = shallow(<GifExpertApp />);    
    });
    test('should load <GifExpertApp /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
})
