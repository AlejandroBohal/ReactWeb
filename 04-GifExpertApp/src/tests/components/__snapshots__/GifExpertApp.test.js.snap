// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp component should load <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
